### ISSUE TITLE
firebase-auth updateUser with providerToLink using "password" as providerId

### ISSUE THREAD

**Issue by AminAtwi-ITXI** [firebase/firebase-admin-node#1592]
Hey I'm trying to link providerData with providerId = "password" using firebase admin SDK
 
 environment :
  * Firebase SDK version:  10.0.2
  * Firebase Product: auth
  * Node.js version: 16.13.0
  * NPM version: 8.1.0

The problem
I'm trying to link providerData using firebase admin SDK in order to link providerData with providerId = "password" .
I'm using updateUser method and pass providerToLink object in order to link providerData.
updateUser method will return this error : 
 - code : auth/invalid-provider-id
 - message : The providerId must be a valid supported provider identifier string.

Relevant Code:

```
await admin
            .auth()
            .updateUser(user.uid, {
                providerToLink: {
                    uid: user.email,
                    email: user.email,
                    displayName: user.displayName,
                    providerId: 'password',
                },
            })
            .catch((error: any) => {
                console.error(`${error}`);
            });
```

**Comment by lahirumaramba** [comment]
If I understand correctly `providerToLink` is used when you want to link a user to a specific provider, such as google.com. If you want to set a password to a user you should be able to set it in the `updateUser` payload. I may have misunderstood your requirement here. Adding @prameshj for more context.

**Comment by prameshj** [comment]
The list of valid provider ids can be found by invoking this API call - https://cloud.google.com/identity-platform/docs/reference/rest/v2/defaultSupportedIdps/list

In this case, is a password change being attempted?

**Comment by tahmassraheem1** [comment]
Have you found the answer?

**Comment by MichaelBorde** [comment]
I don't think you can link a `password` provider directly. You could just call `updateUser` with the email you want (and the password when you know it).

That's odd because you can unlink a `password` provider.

**Comment by sberkaye** [comment]
If you're trying to update user password and add password as the new provider, you can just unlink other provider & set new password in `updateUser`, then sign in with new password and email with `signInWithEmailAndPassword` and pass the new password & email to it.

**Comment by jayenashar** [comment]
what if i'm trying to have multiple emails for one user?  setting `email` on the `UpdateRequest` replaces the email.

**Comment by arnotixe** [comment]
I ran into this yesterday, and adding to what @MichaelBorde mentions, in 2024 and on firebase ~ 10.7.2 you **MUST set a password** for providerData to be populated, **OR** it will be linked after the user logs in with passwordless login (email link).

(+, In the case where a user's email is changed, either by himself or by an admin, the auth user is correctly updated with providerData[] but he's immediately logged out and must re-login with his [new] email. Haven't found a way to change emails on a user without the user being kicked out)

If not setting a password, or until the user has logged in with his new email:
![image](https://github.com/user-attachments/assets/767a609a-22ef-4891-814c-895e156f1b3f)

After setting a password/ logging in:
![image](https://github.com/user-attachments/assets/1d0786f4-24af-4671-aff8-fe8b7de04cb3)
